File: src/main/java/org/jukeboxmc/inventory/DropperInventory.java
package org.jukeboxmc.inventory;

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.blockentity.BlockEntityDropper;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class DropperInventory extends ContainerInventory {

    public DropperInventory( InventoryHolder holder ) {
        super( holder, -1, 9 );
    }

    @Override
    public BlockEntityDropper getInventoryHolder() {
        return (BlockEntityDropper) this.holder;
    }

    @Override
    public @NotNull InventoryType getType() {
        return InventoryType.DROPPER;
    }

    @Override
    public @NotNull ContainerType getWindowTypeId() {
        return ContainerType.DROPPER;
    }
}

File: src/main/java/org/jukeboxmc/inventory/DropperInventory.kt
package org.jukeboxmc.inventory

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType
import org.jukeboxmc.blockentity.BlockEntityDropper

/**
 * @author LucGamesYT
 * @version 1.0
 */
class DropperInventory(holder: InventoryHolder?) : ContainerInventory(holder, -1, 9) {
    override val inventoryHolder: InventoryHolder
        get() = holder as BlockEntityDropper
    override val type: InventoryType
        get() = InventoryType.DROPPER
    override val windowTypeId: ContainerType
        get() = ContainerType.DROPPER
}
